tests/Microsoft.Identity.Test.Unit/PublicApiTests/MtlsPopTests.cs

```diff
@@ -0,0 +1,300 @@
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual("header.payload.signature", result.AccessToken);
+                Assert.AreEqual(Constants.MtlsPoPAuthHeaderPrefix, result.TokenType);
+
+                // Second token acquisition - should retrieve from cache
+                AuthenticationResult secondResult = await app.AcquireTokenForClient(TestConstants.s_scope)
+                    .WithMtlsProofOfPossession()
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual("header.payload.signature", secondResult.AccessToken);
+                Assert.AreEqual(Constants.MtlsPoPAuthHeaderPrefix, secondResult.TokenType);
+                Assert.AreEqual(TokenSource.Cache, secondResult.AuthenticationResultMetadata.TokenSource);
+            }
+        }
+
+        [TestMethod]
+        public void MtlsPop_RegionDiscoveryWithCustomMetadataThrowsException()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var ex = Assert.ThrowsException<MsalClientException>(() => ConfidentialClientApplicationBuilder
+                    .Create(TestConstants.ClientId)
+                    .WithAzureRegion(ConfidentialClientApplication.AttemptRegionDiscovery)
+                    .WithCertificate(s_testCertificate)
+                    .WithInstanceDiscoveryMetadata(File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("CustomInstanceMetadata.json")))
+                    .WithHttpManager(httpManager)
+                    .Build());
+
+                Assert.AreEqual(MsalError.RegionDiscoveryWithCustomInstanceMetadata, ex.ErrorCode);
+                Assert.AreEqual(MsalErrorMessage.RegionDiscoveryWithCustomInstanceMetadata, ex.Message);
+            }
+        }
+
+        [TestMethod]
+        public async Task MtlsPop_KnownRegionAsync()
+        {
+            const string region = "centralus";
+
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddRegionDiscoveryMockHandler(region);
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage(tokenType: "mtls_pop");
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                    .WithCertificate(s_testCertificate)
+                    .WithAzureRegion(region)
+                    .WithHttpManager(httpManager)
+                    .WithExperimentalFeatures()
+                    .BuildConcrete();
+
+                AuthenticationResult result = await app.AcquireTokenForClient(TestConstants.s_scope)
+                    .WithMtlsProofOfPossession()
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                Assert.AreEqual("header.payload.signature", result.AccessToken);
+                Assert.AreEqual(region, result.ApiEvent.RegionUsed);
```

Comment:
Use result.AuthenticationResultMetadata.TokenEndpoint to check the mTLS endpoint.